### ISSUE TITLE
IntelliSense (and tooling in general) is busted for local generators

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#62021]
> Following a conversation on Discord with @CyrusNajmabadi and @sharwell

**Version Used**: VS 17.2, but also VS 17.3 Preview 1

The general issue is that the UX when working with locally referenced generators is pretty broken. If you reference a NuGet package that bundles a source generator, everything works perfectly well: IntelliSense works, you get the files in solution explorer to check out, nothing weird. But, if you have a source generator in the same solution and you just reference that with a local project reference, the UX is pretty poor. Opening the solution the first time after a clean results in IntelliSense being **completely** broken:

![image](https://user-images.githubusercontent.com/10199417/174644920-fd106ce6-7a25-4756-b1f4-324574b1cead.png)

If you build, things actually work, but IntelliSense remains broken until you restart VS. If you apply any changes to the generator and rebuild it, you also need to restart VS for IntelliSense to pick them up. Sometimes it just won't work fine even after restarting, and IntelliSense will just keep showing fake squiggly line over stuff even if building actually does work.

In general, working with local generators is currently a poor developer experience. This is an issue both for testing local generators, and for cases where you just have a generator that's only meant to be used locally, and never actually shipped to NuGet at all.

**Steps to Reproduce**:

1. Clone ComputeSharp: https://github.com/Sergio0694/ComputeSharp
2. Open the solution
3. Go into any of the test projects, and observe red squiggly lines everywhere like in the pic above

We have the same issue in the Microsoft Store app, and I've experienced the same in the MVVM Toolkit as well.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski is this scenario supported?

**Comment by jasonmalinowski** [comment]
Not really, or at least, not without all the caveats that @Sergio0694 is running into here. The issue about you having to reload after changing the generator is tracked by #48083. I can't immediately find the bug for also tracking the you may have to reload after a first build, but there's one out there too for that.

**Comment by jasonmalinowski** [comment]
Closing this since as a duplicate.
